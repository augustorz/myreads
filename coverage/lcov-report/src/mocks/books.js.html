<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/mocks/books.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/mocks</a> books.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">419x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const createBooks = ({ quantity = 1, shelf = 'wantToRead' }) =&gt; (
  Array.from(Array(quantity), (value, index) =&gt; ({
    id: `book-${index}`,
    title: `Author Mock ${index}`,
    authors: [`Title Mock ${index}`],
    imageLinks: {
      thumbnail: `thumbnail_${index}.png`,
    },
    shelf,
  }))
);
&nbsp;
export const createBook = ({
  id = 'book-id-1',
  title = 'title-1',
  authors = ['author-1'],
  imageLinks = {
    thumbnail: 'thumbnail_1.jpg',
  },
  shelf = 'wantToRead',
}) =&gt; ({
  id,
  title,
  authors,
  imageLinks,
  shelf,
});
&nbsp;
export default createBooks;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Apr 01 2019 20:39:37 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
