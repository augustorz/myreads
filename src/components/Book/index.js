import React from 'react';
import PropTypes from 'prop-types';

import BookSelect from '../BookSelect';

const Book = ({ book, onChangeBookShelf }) => (
  <div className="book">
    <div className="book-top">
      <div
        className="book-cover"
        style={{
          width: 128,
          height: 193,
          backgroundImage: `url("${book.imageLinks && book.imageLinks.thumbnail}")`,
        }}
      />
      <BookSelect book={book} onChangeBookShelf={onChangeBookShelf} />
    </div>
    <div className="book-title">{book.title}</div>
    {book.authors && book.authors.map(author => (
      <div className="book-authors" key={author}>{author}</div>
    ))}
  </div>
);

Book.propTypes = {
  book: PropTypes.shape({
    imageLinks: PropTypes.shape({
      thumbnail: PropTypes.string,
    }),
    title: PropTypes.string.isRequired,
    authors: PropTypes.instanceOf(Array),
  }),
  onChangeBookShelf: PropTypes.func.isRequired,
};

Book.defaultProps = {
  book: {
    imageLinks: {
      thumbnail: '',
    },
    authors: [],
  },
};

export default Book;
