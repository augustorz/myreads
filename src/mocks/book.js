const SHELVES = {
  currentlyReading: 'currentlyReading',
  wantToRead: 'wantToRead',
  read: 'read',
};

export default (quantity = 1, shelf = SHELVES.read) => {
  const range = [...Array(quantity)];

  return range.map((value, index) => ({
    authors: [`Authors ${index}`],
    id: `${index}`,
    imageLinks: {
      smallThumbnail: `thumbnail${index}.jpeg`,
      thumbnail: `thumbnail${index}.jpeg`,
    },
    shelf,
    title: `Title ${index}`,
  }));
};
