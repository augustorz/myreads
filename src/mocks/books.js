export const createBooks = ({ quantity = 1, shelf = 'wantToRead' }) => (
  Array.from(Array(quantity), (value, index) => ({
    id: `book-${index}`,
    title: `Author Mock ${index}`,
    authors: [`Title Mock ${index}`],
    imageLinks: {
      thumbnail: `thumbnail_${index}.png`,
    },
    shelf,
  }))
);

export const createBook = ({
  id = 'book-id-1',
  title = 'title-1',
  authors = ['author-1'],
  imageLinks = {
    thumbnail: 'thumbnail_1.jpg',
  },
  shelf = 'wantToRead',
}) => ({
  id,
  title,
  authors,
  imageLinks,
  shelf,
});

export default createBooks;
