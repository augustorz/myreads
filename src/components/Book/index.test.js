import React from 'react';
import { shallow } from 'enzyme';
import { createBook } from '../../mocks/books';

import Book from '.';

describe('Book', () => {
  let wrapper;

  describe('when book has at least one author', () => {
    beforeEach(() => {
      const book = createBook({});

      wrapper = shallow(
        <Book
          book={book}
          onChangeBookShelf={jest.fn()}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('when book has no authors', () => {
    beforeEach(() => {
      const book = createBook({});
      delete book.authors;

      wrapper = shallow(
        <Book
          book={book}
          onChangeBookShelf={jest.fn()}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
