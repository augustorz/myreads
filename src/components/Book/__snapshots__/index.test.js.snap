// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should match snapshot 1`] = `"<div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width:128px;height:193px;background-image:url(&quot;thumbnail_0.png&quot;)\\"></div><div class=\\"book-shelf-changer\\"><select class=\\"book-shelf-select\\"><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option selected=\\"\\" value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 0</div><div class=\\"book-authors\\">Title Mock 0</div></div>"`;
